README.md

```diff
@@ -1,9 +1,28 @@
-# `ConstrOpt`
+# `ConstrOpt` Overview
 This is the official repository for [Leveraging Applicaton Data Constraints to Optimize Database-Backed Web Applications](https://arxiv.org/abs/2205.02954) by Xiaoxuan Liu, Shuxian Wang, Mengzhu Sun, Sharon Lee, Sicheng Pan, Joshua Wu, Cong Yan, Junwen Yang, and Alvin Chueng. 
 Here we present `ConstrOpt`, the first tool that identifies data relationships by analyzing the programs that generate and maintain the persist data. Once identified,`ConstrOpt` leverages the found constraints to optimize the application's physical design and query execution by rewriting queries. Instead of developing a fixed set of predefined rewriting rules, `ConstrOpt` employs an enumerate-test-verify technique to automatically exploit the discovered data constraints to improve query execution. 
-![experiment image](https://github.com/LiuXiaoxuanPKU/ConstrOpt/blob/main/figures/7.2/redmine.pdf)
+## Install
+- Install third party libraries for ruby: under `constr_extractor` folder, run
+``` 
+bundler insatll
+```
+- Install python dependencies: we use [microsoft z3](https://github.com/Z3Prover/z3) solver to prove sql equivalence. use `pip3 install <package name>` to install the following python3 dependencies. E.g., `pip3 install z3-solver`.
```

Comment:
it's better to put all the packages in the requirements.txt and pip install -r requirements.

Reply:
requirements.txt done and pushed